src/hooks/usePreviousImmediate.ts

```diff
@@ -7,6 +8,10 @@ import { useRef, useEffect } from 'react';
  * @returns The previous value
  */
 function usePreviousImmediate<T>(currentValue: T): T | null {
+  warning(
```

Comment:
This will print the warning on each render. 🤔 I would like the warnings to only be logged on the first render. Let's make a `useWarningOnMountInDevelopment` hook. It's a verbose name but it will be super clear. This hook will be for internal purposes and will print the warning only once for each hook. What do you think?